src/arrayMethodSort.js

```diff
@@ -4,8 +4,38 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const initialCompareFunction = (a, b) => {
+      if (String(a) < String(b)) {
+        return -1;
+      }
+
+      return 1;
+    };
+
+    const compare = compareFunction ?? initialCompareFunction;
+
+    let isSwapped = false;
+
+    do {
+      isSwapped = false;
+
+      for (let idx = 0; idx < this.length; idx++) {
+        const result = compare(this[idx], this[idx + 1]);
+
+        if (result > 0) {
```

Comment:
The condition `result > 0` is correct, but it's important to note that if the compare function returns `NaN`, it should be treated as if it returned a positive number, meaning the elements should be swapped.